lib/puppet/application/agent.rb

```diff
@@ -405,7 +404,7 @@ def fingerprint
 
   def onetime(daemon)
     begin
-      exitstatus = daemon.agent.run(:job_id => options[:job_id])
+      exitstatus = daemon.agent.run(:job_id => options[:job_id], :waitforlock => Puppet[:waitforlock] )
```

Comment:
Do we need to pass `:waitforlock` around? From what I see we always use it from `Puppet[:<setting>]`, so it should be available everywhere.